### PR TITLE
promq: add prometheus output format

### PR DESCRIPTION
**Pull request by fpetkovski**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
This PR adds a `prometheus` output format to promq in order to improve the readability of the query result.

Example output
```
$ promq -q "kube_pod_info" --output prometheus  

kube_pod_info{created_by_kind="Node", created_by_name="kind-control-plane", host_ip="172.18.0.2", host_network="true", instance="http://localhost:8080/metrics", namespace="kube-system", node="kind-control-plane", pod="etcd-kind-control-plane", pod_ip="172.18.0.2", priority_class="system-node-critical", uid="b9d1f45f-f984-4d75-9e5b-83fa1ed26104"} => 1 @[1647523773523]
kube_pod_info{created_by_kind="DaemonSet", created_by_name="kindnet", host_ip="172.18.0.2", host_network="true", instance="http://localhost:8080/metrics", namespace="kube-system", node="kind-control-plane", pod="kindnet-5xkll", pod_ip="172.18.0.2", uid="bf51e92b-0da9-48c6-ac6e-010f9f55ea3c"} => 1 @[1647523773523]
```

**Special notes for your reviewer**:
Should we make this output the default?

**Does this PR introduce a user-facing change?**:

```release-note
Add prometheus output to promq
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: promq/prom/output.go [hunk 1]
```diff
@@ -104,6 +105,46 @@ func ToPrettyFormat(res *promql.Result, outputType string, colorized bool) (*str
...
+	}
+	if result.Warnings != nil {
+		for _, e := range result.Warnings {
+			fmt.Println(e)
+		}
+	}
+
+	vector, err := result.Vector()
+	if err != nil {
+		return proto.String(result.String()), nil
```
Comment (logicalhan) [inline promq/prom/output.go:133]: You don't want to also return the error here?

**Review by logicalhan — COMMENTED** [review]
This looks mostly good, just a small comment

File: promq/prom/output.go (hunk 1 above)
Comment (fpetkovski) [reply promq/prom/output.go:133]: This is a bit confusing, the error is returned only when the result is not a `Vector` type. I changed it to a cast to make it more clear what's going on.

Comment (logicalhan) [reply promq/prom/output.go:133]: ah i see, maybe a comment would suffice then?

Comment (fpetkovski) [reply promq/prom/output.go:133]: Sounds good, added a comment for clarity

**Review by logicalhan — APPROVED** [review]
/lgtm
/approve

**Comment by logicalhan** [comment]
Thanks for adding this!

**Pull request merged**